rust/main/agents/relayer/src/merkle_tree/processor.rs

```diff
@@ -89,13 +93,11 @@ pub struct MerkleTreeProcessorMetrics {
 }
 
 impl MerkleTreeProcessorMetrics {
-    pub fn new() -> Self {
+    pub fn new(metrics: &CoreMetrics, origin: &HyperlaneDomain) -> Self {
         Self {
-            max_leaf_index_gauge: IntGauge::new(
-                "max_leaf_index_gauge",
-                "The max merkle tree leaf index",
-            )
-            .unwrap(),
+            max_leaf_index_gauge: metrics
```

Comment:
nit: don't feel too strongly but flagging that `max_leaf_index_gauge` may be nice to rename to `latest_leaf_index` as well for consistency